File: Python3/프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    s_nums = set()
    answer = 0
    
    for i in range(1, len(numbers) + 1):
        for n in permutations(numbers, i):
            s_nums.add(int(''.join(n)))
    
    for n in s_nums:
        if isPrime(n):
            answer += 1
            
    return answer

def isPrime(num):
    if num < 2: return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0: return False
    return True

File: Java/프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    static Set<Integer> nSet = new HashSet<>();
    static boolean[] visited;
    
    public int solution(String numbers) {
        int answer = 0;
        visited = new boolean[numbers.length()];
        dfs(numbers, "", 0);
        for (int n : nSet) {
            if(isPrime(n)) {
                answer ++;
            }
        }
        return answer;
    }
    
    private void dfs(String numbers, String cur, int count) {
        if (!cur.isEmpty()) {
            nSet.add(Integer.parseInt(cur));
        }
        for (int i = 0; i < numbers.length(); i++) {
            if (!visited[i]) {
                visited[i] = true;
                dfs(numbers, cur + numbers.charAt(i), count + 1);
                visited[i] = false;
            }
        }
    }
    
    private boolean isPrime(int num) {
        if (num < 2) return false;
        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
}